extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -99,7 +99,19 @@ export class AmpSidebar extends AMP.BaseElement {
         this.close_();
       }
     });
-    //TODO (skrish, #2712) Add history support on back button.
+
+    // Invisible close button at the end of sidebar for screen-readers.
+    const screenReaderCloseButton = this.document_.createElement('button');
+    // TODO(aghassemi, #4146) i18n
+    screenReaderCloseButton.textContent = 'Close the sidebar';
+    screenReaderCloseButton.classList.add('-amp-screen-reader');
+    // This is for screen-readers only, should not get a tab stop.
+    screenReaderCloseButton.tabIndex = -1;
```

Comment:
generic one doesn't make sense cause these elements could be buttons, paragraphs, links, etc.. depending on the use-case.